src/compile.c

```diff
@@ -0,0 +1,430 @@
+/*
+ * rv32emu is freely redistributable under the MIT License. See the file
+ * "LICENSE" for information on usage and redistribution of this file.
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+
+#include "c2mir/c2mir.h"
+#include "cache.h"
+#include "compile.h"
+#include "decode.h"
+#include "mir-gen.h"
+#include "mir.h"
+#include "riscv_private.h"
+#include "utils.h"
+
+typedef struct {
+    char *code;
+    size_t code_size;
+    size_t curr;
+} jit_item_t;
```

Comment:
I am afraid that `jit_item_t` is a bit misleading. Did you mean that it is a matter of code cache?

Reply:
This is a data structure used by c2mir to retrieve the code string.

Reply:
Maybe we can rename it to `jit_gencode_t`?

Reply:
How about `code_string_t`? By the way, you shall provide more relevant comments.